contracts/src/PriceFeeds/InterchangeablePriceFeed.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+import "../Dependencies/Ownable.sol";
+import "../Interfaces/IPriceFeed.sol";
+import "../BorrowerOperations.sol";
+
+contract InterchangeablePriceFeed is Ownable, IPriceFeed {
+
+    uint256 public lastGoodPrice;
+    
+    IBorrowerOperations public borrowerOperations;
+    IPriceFeed public priceFeed;
+
+    event PriceFeedChanged(address indexed oldPriceFeed, address indexed newPriceFeed);
+    event PriceUpdate(uint256 price, bool isNewOracleFailure);
+
+    constructor(address _owner, IPriceFeed _priceFeed) Ownable(_owner) {
+        priceFeed = _priceFeed;
+    }
+
+    function fetchPrice() external returns (uint256, bool) {
+        return _fetchPrice();
+    }
+
+    function fetchRedemptionPrice() external returns (uint256, bool) {
+        // Use same price for redemption as all other ops in WETH branch
+        return _fetchPrice();
+    }
+
+    function _fetchPrice() internal returns (uint256, bool) {
+        (uint256 price, bool isNewOracleFailure) = priceFeed.fetchPrice();
+
+        emit PriceUpdate(price, isNewOracleFailure);
+
+
+        if (isNewOracleFailure) {
+            return (lastGoodPrice, isNewOracleFailure);
+        } else {
+            lastGoodPrice = price;
+            return (price, isNewOracleFailure);
+        }
+    }
+
+    function setAddresses(address _borrowOperationsAddress) external onlyOwner {
+        borrowerOperations = IBorrowerOperations(_borrowOperationsAddress);
+
+        _renounceOwnership();
```

Comment:
what's the use of the borrower operations? 

Also we should not renounce ownership as we would not be able to update the price feed afterwards

Reply:
good point. borrower operations need to be there to shut down the borrowerOperations